pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -398,3 +413,32 @@ func getEnvironment(m *MachineScope) (*azure.Environment, error) {
 
 	return &env, nil
 }
+
+// getTagList is for getting the merged tag list present in
+// AzureMachineProviderSpec and Infrastructure.Status. Both
+// will have the same tags, unless AzureMachineProviderSpec.Tags
+// has been modified to add/update/delete any tag. Merge will be
+// necessary in delete case, to honour the user-defined tags
+// in Infrastructure.Status and those to newly created resources.
+func getTagList(machineSpecTags, infraStatusTags map[string]string) map[string]*string {
+	if machineSpecTags == nil && infraStatusTags == nil {
+		return nil
+	}
+
+	// infraStatusTags will not be empty when GetCloudEnvironment()
+	// is used for fetching tags in Infrastructure.Status, but
+	// keeping the check for otherwise.
+	if infraStatusTags == nil {
+		infraStatusTags = make(map[string]string)
+	}
+
+	// merge tags present in Infrastructure.Status with
+	// the tags configured in AzureMachineProviderSpec, with
+	// precedence given to those in AzureMachineProviderSpec
+	// for new or updated tags.
+	for k, v := range machineSpecTags {
+		infraStatusTags[k] = v
```

Comment:
You shouldn't mutate the infraStatusTags resource as this is owned by the parent scope, better to just create a new `map[string]*ptr` and iterate over the maps to build up the output map

Reply:
Yes, my bad, did not modify the code when I moved it to a dedicated function.